concrete-integer/src/server_key/crt_parallel.rs

```diff
@@ -0,0 +1,276 @@
+    /// let clear_2 = 14;
+    /// let basis = vec![2, 3, 5];
+    /// // Encrypt two messages
+    /// let mut ctxt_1 = cks.encrypt_crt(clear_1, basis.clone());
+    /// let ctxt_2 = cks.encrypt_crt(clear_2, basis);
+    ///
+    /// // Compute homomorphically a multiplication
+    /// sks.unchecked_add_crt_assign_parallelized(&mut ctxt_1, &ctxt_2);
+    ///
+    /// // Decrypt
+    /// let res = cks.decrypt_crt(&ctxt_1);
+    /// assert_eq!((clear_1 + clear_2) % 30, res);
+    /// ```
+    pub fn unchecked_add_crt_assign_parallelized(
+        &self,
+        ct_left: &mut Ciphertext,
+        ct_right: &Ciphertext,
+    ) {
+        ct_left
+            .ct_vec
+            .par_iter_mut()
+            .zip(&ct_right.ct_vec)
+            .for_each(|(ct_left, ct_right)| {
+                self.key.unchecked_add_assign(ct_left, ct_right);
+            });
+    }
+
+    pub fn unchecked_add_crt_parallelized(
+        &self,
+        ct_left: &Ciphertext,
+        ct_right: &Ciphertext,
+    ) -> Ciphertext {
+        let mut ct_res = ct_left.clone();
+        self.unchecked_add_crt_assign_parallelized(&mut ct_res, ct_right);
+        ct_res
+    }
+
+    /// Computes homomorphically an addition between two ciphertexts encrypting integer values in
+    /// the CRT decomposition.
+    ///
+    /// This checks that the addition is possible. In the case where the carry buffers are full,
+    /// then it is automatically cleared to allow the operation.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use concrete_integer::gen_keys;
+    /// use concrete_shortint::parameters::PARAM_MESSAGE_2_CARRY_2;
+    /// let size = 4;
+    ///
+    /// // Generate the client key and the server key:
+    /// let (cks, sks) = gen_keys(&PARAM_MESSAGE_2_CARRY_2, size);
+    ///
+    /// let clear_1 = 29;
+    /// let clear_2 = 29;
+    /// let basis = vec![2, 3, 5];
+    /// // Encrypt two messages
+    /// let mut ctxt_1 = cks.encrypt_crt(clear_1, basis.clone());
+    /// let mut ctxt_2 = cks.encrypt_crt(clear_2, basis);
+    ///
+    /// // Compute homomorphically a multiplication
+    /// sks.smart_add_crt_assign_parallelized(&mut ctxt_1, &mut ctxt_2);
+    ///
+    /// // Decrypt
+    /// let res = cks.decrypt_crt(&ctxt_1);
+    /// assert_eq!((clear_1 + clear_2) % 30, res);
+    /// ```
+    pub fn smart_add_crt_assign_parallelized(
+        &self,
+        ct_left: &mut Ciphertext,
+        ct_right: &mut Ciphertext,
+    ) {
+        ct_left
+            .ct_vec
+            .par_iter_mut()
+            .zip(&mut ct_right.ct_vec)
+            .for_each(|(block_left, block_right)| {
+                self.key.smart_add_assign(block_left, block_right);
+            });
+    }
+
+    pub fn smart_add_crt_parallelized(
+        &self,
+        ct_left: &mut Ciphertext,
+        ct_right: &mut Ciphertext,
+    ) -> Ciphertext {
+        let mut ct_res = ct_left.clone();
+        self.smart_mul_crt_assign(&mut ct_res, ct_right);
```

Comment:
This calls the wrong function. shouldn't it be calling `smart_add_crt_assign_parallelized` ?

Reply:
you're right, i fixed it now

Reply:
you changed it with `smart_mul_crt_assign_parallelized` when it should be `smart_add_crt_assign_parallelized`

Reply:
thanks for catching that. i wasn't paying attention when i wrote it